common/src/app/common/types/token.cljc

```diff
@@ -558,3 +574,18 @@
   "Predicate if a shadow composite token is a reference value - a string pointing to another reference token."
   [token-value]
   (string? token-value))
+
+(defn update-token-value-references
+  "Recursively update token references within a token value, supporting complex token values (maps, sequences, strings)."
+  [value old-name new-name]
+  (cond
+    (string? value)
+    (str/replace value
+                 (re-pattern (str "\\{" (str/replace old-name "." "\\.") "\\}"))
+                 (str "{" new-name "}"))
+    (map? value)
+    (into {} (map (fn [[k v]] [k (update-token-value-references v old-name new-name)]) value))
```

Comment:
Not very important, but we have a function `app.common.data/update-vals` for this.